src/mqtt/protocol/exchange/exchange_server.c

```diff
@@ -438,6 +470,178 @@ exchange_client_get_msgs_by_key(void *arg, uint64_t key, uint32_t count, nng_msg
 	return 0;
 }
 
+/**
+ * For exchanger, recv_cb is a consumer SDK
+ * TCP/QUIC/IPC/InPROC is at your disposal
+*/
+static void
+ex_query_recv_cb(void *arg)
+{
+	exchange_pipe_t       *p    = arg;
+	exchange_sock_t       *sock = p->sock;
+
+	nni_msg            *msg;
+	size_t              len;
+	uint8_t            *body;
+	nng_aio            *aio;
+	nni_msg            *tar_msg = NULL;
+	int ret;
+
+
+	if (nni_aio_result(&p->ex_aio) != 0) {
+		nni_pipe_close(p->pipe);
+		return;
+	}
+
+
+	msg = nni_aio_get_msg(&p->ex_aio);
+	nni_aio_set_msg(&p->ex_aio, NULL);
+	nni_msg_set_pipe(msg, nni_pipe_id(p->pipe));
+
+	body    = nni_msg_body(msg);
+	len     = nni_msg_len(msg);
+
+	nni_mtx_lock(&sock->mtx);
+	// process query
+	char *keystr = (char *) (body + 4);
```

Comment:
What's your idea about the format of payload in msg

Reply:
TBD